File: 1644C.py
import sys
from functools import lru_cache


def run():
    n, x = map(int, input().split())
    arr = list(map(int, input().split()))

    @lru_cache
    def dp(i, k):
        if i == len(arr):
            return 0
        return max(arr[i] + x + dp(i+1, k-1) if k else 0,
                   arr[i] + dp(i+1, k), arr[i], 0)
    print(*(max(dp(j, i) for j in range(n, -1, -1)) for i in range(n+1)))


def main():
    t = int(input())
    for _ in range(t):
        run()


if __name__ == "__main__":
    try:
        with open("_input.txt", "r") as fin, open("_output.txt", "w") as fout:
            sys.stdin = fin
            sys.stdout = fout
            main()
    except IOError:
        main()

File: 1644C.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long int ll;

const int N = 5e3 + 10;
int arr[N], dp[N][N], n, x;

int asd(int i, int k)
{
    if (i == n)
    {
        return 0;
    }
    if (dp[i][k] != -1)
    {
        return dp[i][k];
    }
    return dp[i][k] = max({k ? arr[i] + x + asd(i + 1, k - 1) : 0,
                           arr[i] + asd(i + 1, k), 0});
}

void run()
{
    scanf("%d%d", &n, &x);
    for (int i = 0; i < n; i++)
    {
        memset(dp + i, -1, sizeof(int) * (n + 1));
    }

    for (int i = 0; i < n; i++)
    {
        scanf("%d", arr + i);
    }
    for (int i = 0; i <= n; i++)
    {
        int mx = 0;
        for (int j = 0; j < n; j++)
        {
            mx = max(mx, asd(j, i));
        }
        printf("%d%c", mx, " \n"[i == n]);
    }
}

int main(int argc, char const *argv[])
{
    // freopen("./_input.txt", "r", stdin);
    // freopen("./_output.txt", "w", stdout);
    int t = 1;
    scanf("%d", &t);
    while (t--)
    {
        run();
    }
    return 0;
}
